s3cmd

```diff
@@ -583,12 +594,79 @@ def cmd_object_del(args):
 
         if not recursive:
             rc = subcmd_object_del_uri(uri_str)
-        else:
+        elif Config().exclude:
             rc = subcmd_batch_del(uri_str = uri_str)
+        else:
+            rc = subcmd_batch_del_iterative(uri_str = uri_str)
         if not rc:
             return rc
     return EX_OK
 
+def subcmd_batch_del_iterative(uri_str = None, bucket = None):
+    """ Streaming version of batch deletion (doesn't realize whole list in memory before deleting). 
+
+    Differences from subcmd_batch_del:
+      - Obeys cfg.max_delete, but if it is >1000, will delete files until we are sure that at least 
+        that many files are to be deleted
+      - Only prints out individual file names if <1000 files are to be deleted
```

Comment:
I understand your point, but what bothers me is to have s3cmd more consistent.
Currently, when you are uploading or syncing, there is the name of each file processed that is displayed. That would be awkward to not have the same behavior for the rm.
Have you tried the "-q" (quiet) option for your use case? Maybe it could be enough to support such an heavy task. And you could still use the exit code of the program to get the success or failure of the operation.